pkg/buildscript/buildscript.go

```diff
@@ -0,0 +1,179 @@
+package buildscript
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/pkg/buildscript/internal/buildexpression"
+	"github.com/ActiveState/cli/pkg/buildscript/internal/raw"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/types"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/go-openapi/strfmt"
+)
+
+// BuildScript is what we want consuming code to work with. This specifically makes the raw presentation private as no
+// consuming code should ever be looking at the raw representation, instead this package should facilitate the use-case
+// of the consuming code through convenience methods that are easy to understand and work with.
+type BuildScript struct {
+	// buildexpression is what we do all our modifications on. We will be planning work to move this to the raw type
```

Comment:
Do we have the JIRA ticket number for this? It may be helpful to reference it here.

Reply:
Will add 👍